grammar.ebnf

```diff
@@ -5,16 +5,23 @@ entry                ::= comment
                       ;
 
 comment              ::= '#' .*;
-section              ::= '[[' keyword ']]';
+section              ::= '[[' __ keyword __ ']]';
+
 __                   ::= [ \t]*;
 NL                   ::= [\r\n]+;
+identifier-head      ::= [a-zA-Z_.?-];
+identifier-tail      ::= [a-zA-Z0-9_.?-];
+keyword-head         ::= [^=|#/{}\[\]()0-9];
+keyword-tail         ::= [^=|#/{}\[\]() ];
+keyword-last         ::= [^=|#/{}\[\]()];
 
-identifier           ::= [a-zA-Z_.?-] ([a-zA-Z0-9_.?-])*;
+identifier           ::= identifier-head (identifier-tail)*;
```

Comment:
I mean - this is the only place where you use it. Since it's not shared with other bits, I'm not sure if there's a value in keeping it as a separate entity.

I believe that it's clear that there are different characters involved since instead of `[...]+` we give `[...] [...]*`.